src/feed_updater.cc

```diff
@@ -0,0 +1,58 @@
+#include "feed_updater.h"
+#include <iostream>
+#include <curl/curl.h>
+#include <chrono>
+#include <thread>
+
+size_t FeedWriteCallback(void* contents, size_t size, size_t nmemb, void* userp) {
+  ((std::string*)userp)->append((char*)contents, size * nmemb);
+  return size * nmemb;
+}
+
+FeedUpdater::~FeedUpdater() {
+  stop();
+}
+
+void FeedUpdater::start() {
+  worker_ = std::thread(&FeedUpdater::run, this);
+}
+
+void FeedUpdater::stop() {
+  running_ = false;
+  if (worker_.joinable()) {
+    worker_.join();
+  }
+}
+
+transit_realtime::FeedMessage& FeedUpdater::getFeed() {
+  return feed_;
+}
+
+void FeedUpdater::run() {
+  while (running_) {
+    if (!downloadFeed()) {
+      std::cerr << "Fehler beim Aktualisieren des Feeds." << std::endl;
+    }
+    std::this_thread::sleep_for(std::chrono::seconds(10));  // Alle 10 Sekunden aktualisieren
```

Comment:
```suggestion
    std::this_thread::sleep_for(std::chrono::seconds(10));
```
I do not think this comment is needed (also not in german)